test/trento_web/controllers/v1/discovery_controller_test.exs

```diff
@@ -32,5 +40,27 @@ defmodule TrentoWeb.V1.DiscoveryControllerTest do
       })
       |> json_response(202)
     end
+
+    test "collect action discards application instance registrations when the associated database does not exists",
+         %{conn: conn} do
+      body =
+        load_discovery_event_fixture("sap_system_discovery_application")
+
+      expect(Trento.Commanded.Mock, :dispatch, fn _ ->
+        {:error, :any_error}
+      end)
```

Comment:
In the end you went the mock way, but I see :smile: no other option available in this case.